repository.go

```diff
@@ -453,3 +454,129 @@ func (r *Repository) Reference(name plumbing.ReferenceName, resolved bool) (
 func (r *Repository) References() (storer.ReferenceIter, error) {
 	return r.s.IterReferences()
 }
+
+// ResolveRevision resolves revision to corresponding hash
+func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, error) {
+	p := revision.NewParserFromString(string(rev))
```

Comment:
Also all this code should not be part of repository, revisions should have a way to resolve themselves (passing them a repository).

Should this resolver return a slice of hashes instead to cope with range revision (when they are supported in the future)?